x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -303,4 +303,66 @@ mod tests {
                 .next()
         );
     }
+
+    #[test]
+    fn assertion_can_fail() {
+        let simulator = simulator::Client::new();
+
+        let owner_key = String::from("owner");
+        let bob_key = Param::Key(Key::Ed25519(String::from("bob")));
+
+        let mut plan = Plan::new(owner_key.clone());
+
+        plan.add_step(Step::create_key(Key::Ed25519(owner_key)));
+
+        plan.add_step(Step {
+            endpoint: Endpoint::Key,
+            method: "key_create".into(),
+            params: vec![bob_key.clone()],
+            max_units: 0,
+            require: None,
+        });
+
+        let counter1_id = plan.add_step(Step {
+            endpoint: Endpoint::Execute,
+            method: "program_create".into(),
+            max_units: 1000000,
+            params: vec![Param::String(PROGRAM_PATH.into())],
+            require: None,
+        });
+        plan.add_step(Step {
+            endpoint: Endpoint::Execute,
+            method: "initialize_address".into(),
+            max_units: 1000000,
```

Comment:
While writing this PR, @richardpringle (thanks!) found that I broke assertions. Having this test would have caught it